rpc/rpcserver/server.go

```diff
@@ -796,18 +796,64 @@ func (s *walletServer) SweepAccount(ctx context.Context, req *pb.SweepAccountReq
 		return nil, translateError(err)
 	}
 
-	var txBuf bytes.Buffer
-	txBuf.Grow(tx.Tx.SerializeSize())
-	err = tx.Tx.Serialize(&txBuf)
-	if err != nil {
-		return nil, translateError(err)
+	sweepTxs := make([]*txauthor.AuthoredTx, 0)
+	maxTxSize := s.wallet.ChainParams().MaxTxSize
+	if tx.EstimatedSignedSerializeSize > maxTxSize {
```

Comment:
I don't think the gRPC handler is the correct place for this logic.  The wallet should be returning multiple transactions directly instead.  `NewUnsignedTransaction` can be modified to never create transactions that exceed the maximum size, and a new wallet method can be introduced to sweep an account and return a `[]*txauthor.AuthoredTx`.